ex04.ex

```diff
@@ -75,7 +80,25 @@ defmodule Ex04 do
   return value will be the thing you have to manipulate.
   """
 
-  def even_odd . . . "your code"
+  def even_odd (list) do
```

Comment:
## -4

This is a big, complex function. It duplicates a lot of code, and it also duplicates the processing of the list.

I'd like to see something more like:

~~~ elixir
  def even_odd(list) do
    reduce(list, { [], [] }, &partition_even_odd/2)
    |> reverse_two_lists
  end

  defp partition_even_odd(x, { es, os }) when Integer.is_even(x), do: {[x | es], os}
  defp partition_even_odd(x, { es, os }) when Integer.is_odd(x),  do: {es, [x | os]}

  defp reverse_two_lists({ evens, odds }) do
    { reverse(evens), reverse(odds) }
  end
~~~